resources/upload.rb

```diff
@@ -0,0 +1,54 @@
+# encoding: utf-8
+
+provides :compliance_upload
+resource_name :compliance_upload
+
+property :path, String
+property :owner, String, required: true
+property :server, [String, URI, nil]
+property :insecure, [TrueClass, FalseClass], default: false
+property :overwrite, [TrueClass, FalseClass], default: true
+property :refresh_token, String
+property :profile_name, String
+
+default_action :upload
+
+# upload profile to compliance server
+action :upload do
+  converge_by 'run profile validation checks' do
+    load_inspec_libs
+    raise 'Path to profile archive not specified' if path.nil?
+    raise "Profile archive file #{path} does not exist." unless ::File.exist?(path)
+    profile = Inspec::Profile.for_target(path, {})
+    error_count = 0
+    lambda { |msg|
```

Comment:
i think that lambda was just cruft leftover. i'll remove and test in a bit.   oh ya, good point on the errors.  i'll go through and update those! :)